### PR TITLE
Http methods update for static files and pages

### PR DESCRIPTION
**Pull request by huv1k**
This PR only allows `GET` and `HEAD` for `static` files and reverts `POST` on `pages` to return `404`.

This PR implements the following:
- `404` on missing assets, pages, API routes, etc (*any missing route*)
- `405` on non-`GET`/`HEAD` to `pages/`, `static/`, and `public/`
- All requests are processed as-is for matched API routes

### CODE REVIEW

File: packages/next-server/server/next-server.ts [hunk 1]
```diff
@@ -653,6 +657,11 @@ export default class Server {
     path: string,
     parsedUrl?: UrlWithParsedQuery
   ): Promise<void> {
+    if (req.method !== 'GET' && req.method !== 'HEAD') {
+      res.statusCode = 405
```
Comment (Timer) [inline packages/next-server/server/next-server.ts]: We need to set an appropriate `Allow` header specifiying `GET` and `HEAD`.

File: packages/next-server/server/next-server.ts [hunk 2]
```diff
@@ -441,6 +441,10 @@ export default class Server {
     query: ParsedUrlQuery = {},
     parsedUrl?: UrlWithParsedQuery
   ): Promise<void> {
+    if (req.method !== 'GET' && req.method !== 'HEAD') {
+      return this.render404(req, res, parsedUrl)
```
Comment (Timer) [inline packages/next-server/server/next-server.ts]: We shouldn't be returning an explicit `404`, the previous implementation just didn't match -- if we're responding on behalf of the user then it should be a `405`. I still think this should follow the old behavior.

**Review by Timer — CHANGES_REQUESTED** [review]

**Review by Timer — APPROVED** [review]
I tweaked this to `405` on pages instead of `501` per Slack conversations.

File: test/integration/production/test/index.test.js [hunk 3]
```diff
@@ -80,18 +80,52 @@ describe('Production Usage', () => {
...
+      }
+    })
+
     it('should render 404 for dotfiles in /static', async () => {
       const url = `http://localhost:${appPort}/static/.env`
       const res = await fetch(url)
       expect(res.status).toBe(404)
     })
 
+    it('should retrun 405 method on static then GET and HEAD', async () => {
```
Comment (ijjk) [inline test/integration/production/test/index.test.js]:
```suggestion
    it('should return 405 method on static then GET and HEAD', async () => {
```

**Review by ijjk — APPROVED** [review]

**Pull request merged**